File: remove_auditlog.py
import os
import re
from pathlib import Path

def remove_auditlog_blocks(content):
    """Remove all auditLog.create blocks from content"""
    
    # Pattern 1: Try-catch blocks containing ONLY auditLog
    pattern1 = r'\/\/[^\n]*(?:[Cc]réer|[Cc]reate).*?(?:audit|log d\'audit)[^\n]*\n\s*try\s*\{[^}]*?await\s+(?:prisma|tx)\.auditLog\.create\s*\([^;]*?\);?\s*[^}]*?\}\s*catch\s*\([^)]*\)\s*\{[^}]*?\}\s*'
    
    # Pattern 2: Standalone auditLog with comment
    pattern2 = r'\/\/[^\n]*(?:[Cc]réer|[Cc]reate).*?(?:audit|log d\'audit)[^\n]*\n\s*(?:\/\*\s*)?await\s+(?:prisma|tx)\.auditLog\.create\s*\([^;]*?\);?\s*(?:\*\/)?'
    
    # Remove try-catch blocks first
    modified = re.sub(pattern1, '\n', content, flags=re.DOTALL)
    
    # Then remove standalone calls
    modified = re.sub(pattern2, '\n', modified, flags=re.DOTALL)
    
    # Clean up excessive blank lines
    modified = re.sub(r'\n{3,}', '\n\n', modified)
    
    return modified

def process_file(filepath):
    """Process a single TypeScript file"""
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has auditLog references
        if 'auditLog.create' not in content:
            return False
        
        original_count = content.count('auditLog.create')
        modified = remove_auditlog_blocks(content)
        new_count = modified.count('auditLog.create')
        
        if new_count < original_count:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(modified)
            removed = original_count - new_count
            print(f"✅ {filepath.relative_to(Path.cwd())}: removed {removed} auditLog blocks")
            return True
        
        return False
    except Exception as e:
        print(f"❌ Error processing {filepath}: {e}")
        return False

def main():
    """Main function to process all TypeScript files in app/api"""
    api_dir = Path('app/api')
    
    if not api_dir.exists():
        print("❌ Directory app/api not found!")
        return
    
    ts_files = list(api_dir.rglob('*.ts'))
    total_files = 0
    
    print(f"\n🔍 Found {len(ts_files)} TypeScript files in app/api\n")
    
    for filepath in ts_files:
        if process_file(filepath):
            total_files += 1
    
    print(f"\n✅ Total: {total_files} files modified\n")

if __name__ == '__main__':
    main()

File: remove-auditlog.js
const fs = require('fs');
const path = require('path');
const glob = require('glob');

// Find all .ts files in app/api
const files = glob.sync('app/api/**/*.ts', { cwd: process.cwd() });

let totalFiles = 0;
let totalRemoved = 0;

files.forEach(file => {
    const fullPath = path.join(process.cwd(), file);
    let content = fs.readFileSync(fullPath, 'utf8');
    const originalContent = content;

    // Pattern to match auditLog.create blocks with their surrounding try-catch or standalone
    // This regex matches from the comment to the closing });
    const patterns = [
        // Pattern 1: Match try-catch blocks containing only auditLog
        /\s*\/\/[^\n]*audit[^\n]*\n\s*try\s*\{[^}]*await\s+(?:prisma|tx)\.auditLog\.create\s*\([^)]*\)[^}]*\}\s*catch[^}]*\{[^}]*\}\s*/gi,

        // Pattern 2: Match standalone auditLog.create with comment
        /\s*\/\/[^\n]*(?:audit|log d'audit)[^\n]*\n\s*await\s+(?:prisma|tx)\.auditLog\.create\s*\(\{(?:[^{}]|\{[^{}]*\})*\}\s*\)\s*;?\s*/gi,

        // Pattern 3: Match auditLog.create inside try blocks (without the try-catch)
        /\s*await\s+(?:prisma|tx)\.auditLog\.create\s*\(\{(?:[^{}]|\{[^{}]*\})*\}\s*\)\s*;?\s*/gi,
    ];

    let modified = false;
    let removeCount = 0;

    patterns.forEach(pattern => {
        const matches = content.match(pattern);
        if (matches) {
            removeCount += matches.length;
            content = content.replace(pattern, '\n');
            modified = true;
        }
    });

    // Clean up multiple consecutive blank lines
    content = content.replace(/\n{3,}/g, '\n\n');

    if (modified && content !== originalContent) {
        fs.writeFileSync(fullPath, content, 'utf8');
        console.log(`✅ ${file}: removed ${removeCount} auditLog blocks`);
        totalFiles++;
        totalRemoved += removeCount;
    }
});

console.log(`\n========================================`);
console.log(`✅ Total: ${totalFiles} files modified`);
console.log(`✅ Total: ${totalRemoved} auditLog blocks removed`);
console.log(`========================================\n`);
